provider/k8s/release.go

```diff
@@ -334,19 +334,20 @@ func (p *Provider) releaseTemplateCA(a *structs.App, ca *v1.Secret) ([]byte, err
 }
 
 func (p *Provider) releaseTemplateIngress(a *structs.App, ss manifest.Services, opts structs.ReleasePromoteOptions) ([]byte, error) {
-	ans, err := p.Engine.IngressAnnotations(a.Name)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
 	idles, err := p.Engine.AppIdles(a.Name)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
 	items := [][]byte{}
 
-	for _, s := range ss {
+	for i := range ss {
+		s := ss[i]
+		ans, err := p.Engine.IngressAnnotations(s.Certificate.Duration)
```

Comment:
I can see you defined rack param and also service wise parameter for cert renewal.
For service, if it is not defined but rack cert param is defined, then I think we should use it as the default value.

Reply:
hey @nightfury1204 I thought the rack value is used by default for services.

Reply:
not sure can you check please?

Reply:
yeah, I checked, I was just double checking I you've seen anything that I haven't.

Reply:
don't you think we should use rack cert renewal period as default?

Reply:
The default is the rack renewal period. If there's nothing setup for service, it uses the default rack period.